tests/unit/amo/components/TestCollection.js

```diff
@@ -0,0 +1,445 @@
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    const wrapper = renderComponent({ store });
+
+    const id = 'error-handler-id';
+    const error = createApiError({
+      response: { status: 500 },
+      apiURL: 'https://some/api/endpoint',
+      jsonResponse: { message: 'Nope.' },
+    });
+    store.dispatch(setError({ id, error }));
+    const capturedError = store.getState().errors[id];
+    expect(capturedError).toBeTruthy();
+
+    const errorHandler = new ErrorHandler({
+      id, dispatch: sinon.stub(), capturedError,
+    });
+
+    fakeDispatch.reset();
+    wrapper.setProps({ errorHandler });
+
+    sinon.assert.notCalled(fakeDispatch);
+  });
+
+  it('dispatches fetchCollection when location pathname has changed', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    // We need a collection for this test case.
+    store.dispatch(loadCollection({
+      addons: createFakeCollectionAddons(),
+      detail: createFakeCollectionDetail(),
+    }));
+
+    const errorHandler = createStubErrorHandler();
+    const slug = 'collection-slug';
+    const user = 'some-user';
+    const page = 123;
+
+    const location = {
+      pathname: `/collections/${user}/${slug}/`,
+      query: { page },
+    };
+
+    const newSlug = 'other-collection';
+    const newLocation = {
+      ...location,
+      pathname: `/collections/${user}/${newSlug}/`,
+    };
+
+    const wrapper = renderComponent({
+      errorHandler,
+      location,
+      params: { slug, user },
+      store,
+    });
+    fakeDispatch.reset();
+
+    // This will trigger the componentWillReceiveProps() method.
+    wrapper.setProps({
+      location: newLocation,
+      params: { slug: newSlug, user },
+    });
+
+    sinon.assert.callCount(fakeDispatch, 1);
+    sinon.assert.calledWith(fakeDispatch, fetchCollection({
+      errorHandlerId: errorHandler.id,
+      page,
+      slug: newSlug,
+      user,
+    }));
+  });
+
+  it('dispatches fetchCollectionPage when page has changed', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    // We need a collection for this test case.
+    store.dispatch(loadCollection({
+      addons: createFakeCollectionAddons(),
+      detail: createFakeCollectionDetail(),
+    }));
+
+    const page = 123;
+    const location = { query: {} };
+    const newLocation = { query: { page } };
+
+    const errorHandler = createStubErrorHandler();
+    const slug = 'collection-slug';
+    const user = 'some-user';
+
+    const wrapper = renderComponent({
+      errorHandler,
+      location,
+      params: { slug, user },
+      store,
+    });
+    fakeDispatch.reset();
+
+    // This will trigger the componentWillReceiveProps() method.
+    wrapper.setProps({ location: newLocation });
```

Comment:
Could we dispatch here instead as well?

Reply:
`location` and `params` are not handled by redux.